src/treegen.cpp

```diff
@@ -183,6 +197,17 @@ treegen::error make_ltree(MMVManip &vmanip, v3s16 p0,
 	std::stack <core::matrix4> stack_orientation;
 	std::stack <v3f> stack_position;
 
+	// fill NodeDefs with unique id if requested
+	if (tree_definition.enable_unique_ids) {
+		u8 unique_id = 0;
+		PcgRandom r((((p0.X << 16) + p0.Y) << 16) + p0.Z);
```

Comment:
I'd imagine the compiler will optimise this better without the parenthesis; i.e. p0.X << 32 + p0.Y << 16 + p0.Z

Also, all modern compilers will optimise pot2 multiplications to shifts, but I guess in this case it's clearer the way it is